doc/man7/provider-asymcipher.pod

```diff
@@ -181,6 +181,13 @@ algorithms:
 
 =item "pad-mode" (B<OSSL_ASYM_CIPHER_PARAM_PAD_MODE>) <integer>
 
+The type of padding to be used. The interpretation of this value will depend
+on the algorithm in use. The default provider understands these RSA padding
+modes: 1 (RSA_PKCS1_PADDING), 2 (RSA_SSLV23_PADDING), 3 (RSA_NO_PADDING),
+4 (RSA_PKCS1_OAEP_PADDING), 5 (RSA_X931_PADDING), 6 (RSA_PKCS1_PSS_PADDING) and
+7 (RSA_PKCS1_WITH_TLS_PADDING). See L<EVP_PKEY_CTX_set_rsa_padding(3)> for
```

Comment:
Do we want to document the numbers? I think it should just list  the defines.

Reply:
This provides part of the public API to providers. Providers don't need to link against libcrypto at all if they don't want to and may not include our headers (or even be written in C - as long as they conform to the C calling convention). For that reason we've tended to document the numbers in the man7 provider documentation (at least we have for params, and if we do it there it makes sense to me to do it here too).